### PR TITLE
Add BTT Octopus support

### LINKED ISSUE #248: Add example config for BTT Octopus
Add an example configuration file for the BTT Octopus board.

### PR DESCRIPTION
**Pull request by stephenlawrence**
Fixes #248

### CODE REVIEW

**Comment by julianschill** [comment]
Please check the changes I did to the SKR configuration in the probe section: https://github.com/VoronDesign/Voron-2/commit/c541c1333bf9f59e0238ebd5155feb760e6465df
This also affects the Octopus config.

**Comment by cbc02009** [comment]
> 
> 
> Please check the changes I did to the SKR configuration in the probe section: [c541c13](https://github.com/VoronDesign/Voron-2/commit/c541c1333bf9f59e0238ebd5155feb760e6465df)
> This also affects the Octopus config.

the Octopus uses an optopcoupler, so that isn't relevant here.

As a suggestion, because of how the optocoupler interferes with a lot of the probes from the sourcing guide, I would recommend either having two commented out probe sections (one for SENSOR and one for DIAG7 w/ BAT85 for instance) and allow the user to uncomment the relevant one.

Also the pin for probe needs to be ~!PB7 for an NC probe to work correctly. The triggered state (connection broken) turns off the mosfet, and the pulldown resistor pulls the signal to ground. the open state (connection made) allows current to flow through the transistor in the optocoupler and pulls the signal high.

**Comment by cbc02009** [comment]
```
#####################################################################
# 	Probe
#####################################################################

[probe]
##	Inductive Probe
##	This probe is not used for Z height, only Quad Gantry Leveling
##	Connected to SENSOR or DIAG7 (with BAT85)
##	This can be connected in two positions depending on which probe
##    You're using. Uncomment the correct pin below
# pin: ~!PB7 # SENSOR port. Remove ! if probe is NO
# pin: PG15  # DIAG7 port. NEEDS BAT85! Change to !PG15 if probe is NO. 
x_offset: 0
y_offset: 25.0
z_offset: 0
speed: 10.0
samples: 3
samples_result: median
sample_retract_dist: 3.0
samples_tolerance: 0.006
samples_tolerance_retries: 3
```
 I was thinking something like this

**Comment by cbc02009** [comment]
@stephenlawrence do you want me to take over this pull request?

**Comment by cbc02009** [comment]
What still needs to be done for this pull request?

**Comment by stephenlawrence** [comment]
> What still needs to be done for this pull request?

let me grab my in-use config and see what I ended up changing along the way.

**Comment by stephenlawrence** [comment]
> What still needs to be done for this pull request?

There are a number of changes I made while getting my 2.4 up and running properly. Mostly pin changes. I am thinking I should merge those changes?

**Comment by cbc02009** [comment]
> 
> 
> > What still needs to be done for this pull request?
> 
> There are a number of changes I made while getting my 2.4 up and running properly. Mostly pin changes. I am thinking I should merge those changes?

yeah, probably a good idea.

I do think adding the note in the comment above about how to wire it depending on whether you have the diode or not makes sense, given the questions people ask on discord.

**Comment by stephenlawrence** [comment]
> > > What still needs to be done for this pull request?
> > 
> > 
> > There are a number of changes I made while getting my 2.4 up and running properly. Mostly pin changes. I am thinking I should merge those changes?
> 
> yeah, probably a good idea.
> 
> I do think adding the note in the comment above about how to wire it depending on whether you have the diode or not makes sense, given the questions people ask on discord.

Updated.

File: firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg [hunk 1]
```diff
@@ -0,0 +1,646 @@
...
+enable_pin: !PG5
+rotation_distance: 40
+gear_ratio: 80:16
+microsteps: 16
+endstop_pin: PG10
+##  Z-position of nozzle (in mm) to z-endstop trigger point relative to print surface (Z0)
+##  (+) value = endstop above Z0, (-) value = endstop below
+##	Increasing position_endstop brings nozzle closer to the bed
+##  After you run Z_ENDSTOP_CALIBRATE, position_endstop will be stored at the very end of your config
+#position_endstop: -0.5
```
Comment (cbc02009) [inline firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:208]: I think this needs to be uncommented in the default config or else klipper will throw an error.

File: firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg [hunk 2]
```diff
@@ -0,0 +1,646 @@
...
+#####################################################################
+
+[probe]
+##	Inductive Probe
+##	This probe is not used for Z height, only Quad Gantry Leveling
+##	and is Connected to SENSOR or DIAG7 (with BAT85)
+##	This can be connected in two positions depending on which probe
+##  you are using. Uncomment the correct pin below
+# pin: ~!PB7 # SENSOR port. Remove ! if probe is NO
+# pin: ^PG15  # DIAG7 port. NEEDS BAT85! Change to !^PG15 if probe is NO.
```
Comment (cbc02009) [inline firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:366]: The rest of the probe section seems to have been removed by accident.

File: firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg [hunk 3]
```diff
@@ -0,0 +1,646 @@
...
+    G91                            ; relative positioning
+    G0 Z1.00 X20.0 Y20.0 F20000    ; move nozzle to remove stringing
+    TURN_OFF_HEATERS
+    M107                           ; turn off fan
+    G1 Z2 F3000                    ; move nozzle up 2mm
+    G90                            ; absolute positioning
+    G0  X125 Y250 F3600            ; park nozzle at rear
+    BED_MESH_CLEAR
+    
+[gcode_macro PAUSE]
```
Comment (cbc02009) [inline firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:569]: I think PAUSE and RESUME are normally part of fluidd.cfg or mainsail.cfg, so including them in the default config could throw errors for people running pre-made images (which is probably the majority of the install base currently)

File: firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg [hunk 4]
```diff
@@ -0,0 +1,646 @@
...
+## 	Thermistor Types
+##   "EPCOS 100K B57560G104F"
+##   "ATC Semitec 104GT-2"
+##   "NTC 100K beta 3950"
+##   "Honeywell 100K 135-104LAG-J01"
+##   "NTC 100K MGB18-104F39050L32" (Keenovo Heater Pad)
+##   "AD595"
+##   "PT100 INA826"
+
+[virtual_sdcard]
```
Comment (cbc02009) [inline firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:641]: Same with these lines. I think everything below 640 should be removed from the default config file.

**Review by cbc02009 — CHANGES_REQUESTED** [review]
Thanks for the updates. just a couple things I think need to be changed.

File: firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg [hunk 5]
```diff
@@ -0,0 +1,646 @@
...
+stealthchop_threshold: 0
+
+#####################################################################
+# 	Bed Heater
+#####################################################################
+
+[heater_bed]
+##	SSR Pin - HE1
+##  Thermistor - TB
+heater_pin: PD14
```
Comment (mytran) [inline firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:347]: I think this should be PA1 for the SSR

Comment (cbc02009) [reply firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:347]: HE1 is actually PA3, but good catch.

I don't see the point in using BED_OUT for the SSR. it's extra wiring for no gain.

File: firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+dir_pin: PE3
+enable_pin: !PD4
+##	Update value below when you perform extruder calibration
+##	If you ask for 100mm of filament, but in reality it is 98mm:
+##	rotation_distance = <previous_rotation_distance> * <actual_extrude_distance> / 100
+##  22.6789511 is a good starting point
+rotation_distance: 22.6789511	#Bondtech 5mm Drive Gears
+##	Update Gear Ratio depending on your Extruder Type
+##	Use 50:17 for Afterburner/Clockwork (BMG Gear Ratio)
+##	Use 80:20 for M4, M3.1
```
Comment (CadmonkeyFPV) [inline firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:311]: Add LGX rotation distance: 8 according to BMG, 7.780624 according to my config (hot end removed)

File: firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg [hunk 7]
```diff
@@ -0,0 +1,646 @@
...
+
+## MISC Pins
+## SENSOR              PB7 (make sure to set jumper correctly)
+##===================================================================
+
+[mcu]
+##	Obtain definition by "ls -l /dev/serial/by-id/" then unplug to verify
+##--------------------------------------------------------------------
+serial: /dev/serial/by-id/{REPLACE WITH YOUR SERIAL}
+##--------------------------------------------------------------------
```
Comment (cbc02009) [inline firmware/klipper_configurations/Octopus/Voron2_Octopus_Config.cfg:95]: need to add `restart_method: command` the the [mcu] section for resets to work properly on some boards.

**Comment by cbc02009** [comment]
Octopus config merged, this can be closed out.

**Comment by raymondh2** [comment]
Support added in #270

**Pull request closed without merging**